workflow-ui/container-android/src/androidTest/java/com/squareup/workflow1/ui/modal/test/ModalViewContainerLifecycleActivity.kt

```diff
@@ -56,20 +48,15 @@ internal class ModalViewContainerLifecycleActivity : AbstractLifecycleTestActivi
     ModalViewContainer.binding<TestModals>(),
     BaseRendering,
     leafViewBinding(LeafRendering::class, lifecycleLoggingViewObserver { it.name }),
-    ManualScreenViewFactory(RecurseRendering::class) { initialRendering,
-      initialViewEnvironment,
-      contextForNewView, _ ->
-      FrameLayout(contextForNewView).also { container ->
-        val stub = WorkflowViewStub(contextForNewView)
+    forBuiltView<RecurseRendering> { _, environment, context, _ ->
+      val stub = WorkflowViewStub(context)
+      val frame = FrameLayout(context).also { container ->
         container.addView(stub)
-        container.bindShowRendering(
-          initialRendering,
-          initialViewEnvironment
-        ) { rendering, env ->
-          stub.show(asScreen(TestModals(listOf(rendering.wrapped))), env)
-        }
       }
-    },
+      ScreenViewHolder(environment, frame) { rendering, viewEnvironment ->
+        stub.show(asScreen(TestModals(listOf(rendering.wrapped))), viewEnvironment)
```

Comment:
Compare this with #703, where we were forced to cast `(stub as WorkflowViewStub)`